test/unit/SubscribedStream.test.js

```diff
@@ -0,0 +1,144 @@
+import assert from 'assert'
+import sinon from 'sinon'
+import { PublishRequest, StreamMessage } from 'streamr-client-protocol'
+import StreamrClient from '../../src/StreamrClient'
+import SubscribedStream from '../../src/SubscribedStream'
+import Signer from '../../src/Signer'
+
+describe('SubscribedStream', () => {
```

Comment:
Comment applies to several places in this file: you're testing (and manipulating) implementation _internals_. Is there a way to test _behavior_ instead (using the _public API_ of the class)?

Example of antipattern:
- When `StreamrClient` has option `verifySignatures: 'auto'`, internal flag `verifySignatures` has value `undefined`

Example of behavioral pattern:
- When `StreamrClient` has option `verifySignatures: 'auto'`, message signature is verified if present

How to observe behavior, then?
- Public API
- Collaborator calls (stubbed)